### PR TITLE
Add new template for simple windows vm with diagnostics

### PR DESCRIPTION
**Pull request by sbtron**

### CODE REVIEW

File: 101-simple-windows-vm-monitoring-diagnostics/README.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+
+Built by: [sbtron](https://github.com/sbtron)
+
+This template allows you to deploy a simple Windows VM along with the diagnostics extension which enables monitoring and diagnostics for the VM. For more information see [Azure Diagnostics Extension in a resource manager template](http://azure.microsoft.com/documentation/articles/virtual-machines-extensions-diagnostics-windows-template). 
+
+Below are the parameters that the template expects: 
+
+| Name   | Description    |
```
Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/README.md]: This table isn't needed anymore. It's already being rendered on Azure.com

File: 101-simple-windows-vm-monitoring-diagnostics/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# A Windows VM with the Azure diagnostics extension which enables monitoring and diagnostics capabilities 
+
+<a href="https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2F101-simple-windows-vm-monitoring-diagnostics%2Fazuredeploy.json" target="_blank">
+    <img src="http://azuredeploy.net/deploybutton.png"/>
+</a>
+
+Built by: [sbtron](https://github.com/sbtron)
```
Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/README.md]: This should be removed. This information is captured in the Github history and metadata file

Comment (sbtron) [inline 101-simple-windows-vm-monitoring-diagnostics/README.md]: should I also remove the deploy button?

Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/README.md]: Deploy button is fine for anyone deploying from here. For other stuff below, the ACOM team will render the readme on Azure.com so a lot of this stuff ends up being doubled but they can exclude the deploy button from the readme.

File: 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        "description": "Username for the Virtual Machine."
+      }
+    },
+    "adminPassword": {
+      "type": "securestring",
+      "metadata": {
+        "description": "Password for the Virtual Machine."
+      }
+    },
+    "dnsNameForPublicIP": {
```
Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json]: this causes a lot of failures, I suggest using the uniqueString function in ARM to generate this

File: 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        "2012-R2-Datacenter"
+      ],
+      "metadata": {
+        "description": "The Windows version for the VM. This will pick a fully patched image of this given Windows version. Allowed values: 2008-R2-SP1, 2012-Datacenter, 2012-R2-Datacenter."
+      }
+    },
+      "diagnosticsStorageAccountName": {
+          "type": "string",
+          "metadata": {
+        "description": "The name of the storage account to which diagnostics data will be transfered."
```
Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json]: Same comment as dns name above, this also causes a lot of failures. Something like 13 char unique string from uniqueString() + a suffix will work

File: 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+      }
+      },
+      "diagnosticsStorageResourceGroup": {
```
Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json]: existing prefix for this as well

File: 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json [hunk 6]
```diff
@@ -0,0 +1,228 @@
...
+      "wadcfgxstart": "[concat(variables('wadlogs'), variables('wadperfcounters1'), variables('wadperfcounters2'), '<Metrics resourceId=\"')]",
+      "wadmetricsresourceid": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name , '/providers/', 'Microsoft.Compute/virtualMachines/')]",
+      "wadcfgxend": "\"><MetricAggregation scheduledTransferPeriod=\"PT1H\"/><MetricAggregation scheduledTransferPeriod=\"PT1M\"/></Metrics></DiagnosticMonitorConfiguration></WadCfg>"
+  },
+  "resources": [
+    {
+      "type": "Microsoft.Storage/storageAccounts",
+      "name": "[parameters('newStorageAccountName')]",
+      "apiVersion": "2015-05-01-preview",
```
Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json]: I suggest making the api version a variable and also using the GA api 2015-06-15

File: 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json (hunk 3 above)
Comment (sbtron) [inline 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json]: can you share a way to do unique DNS. It needs to be unique across the region I assume? :)

Comment (singhkay) [inline 101-simple-windows-vm-monitoring-diagnostics/azuredeploy.json]: See here https://github.com/Azure/azure-quickstart-templates/blob/bfb0c40965969a4884ccee65ff75c7ee13e3195b/301-2fe-linux-lb80-ssh-1be-win-nsg-rdp-datadisk-ssd/azuredeploy.json#L90

The string generated should be sufficiently unique

**Comment by sbtron** [comment]
All requested changes have been made please review and merge.

**Comment by singhkay** [comment]
@sbtron Can you change the name of the template to 201-vm-monitoring-diagnostics-extension. This is a bit more involved to understand than 101-simple-windows-vm and also with the Github's 27 character width on folder names column, it help the scenario name stand out for easier identification

**Pull request merged**